scanner/scanner.go

```diff
@@ -14,13 +14,18 @@ import (
 // ScannerAble is implemented by zbuf.Readers that provide an optimized
 // implementation of the Scanner interface.
 type ScannerAble interface {
-	NewScanner(ctx context.Context, filterExpr ast.BooleanExpr, s nano.Span) (Scanner, error)
+	NewScanner(ctx context.Context, f filter.Filter, filterExpr ast.BooleanExpr, s nano.Span) (Scanner, error)
+}
+
+// A ScannerStatsAble generates scanner statistics.
+type ScannerStatsAble interface {
+	Stats() *ScannerStats
```

Comment:
This is an odd name. Statter is more conventional.

Reply:
Bah. The method is Stats, not Stat, so make that Statser.

Reply:
step by step, you are working out how I came to 'ScannerStatsAble'. :)